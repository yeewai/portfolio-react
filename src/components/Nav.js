import React from 'react';
import { Link } from 'react-router-dom';

export const Nav = ({ match }) => {
    const whichPage = ["/art", "/about", "/code"].find( p => match.url.match(`^${p}`));

    return (
        <nav className="mainnav row justify-content-around animated slideInDown">
            <div className={`col item ${whichPage === "/art" ? 'selected' : ''}`} id="art" >
                <Link to="/art" className="nav"><h2>Art</h2></Link>
            </div>
            <div className={`col-2 item ${whichPage === "/about" ? 'selected' : ''}`} id="about" >
                <Link to="/about" className="nav"><h2>Yee Wai</h2></Link>
            </div>
            <div className={`col item ${whichPage === "/code" ? 'selected' : ''}`} id="code" >
                <Link to="/code"className="nav"><h2>Code</h2></Link>
            </div>
        </nav>
    )
};

export default Nav;
