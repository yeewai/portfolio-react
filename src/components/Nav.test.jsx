import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Nav } from './Nav';

Enzyme.configure({ adapter: new Adapter() });

describe("Nav", () => {
    it('renders', () => {
      const wrapper = shallow(<Nav match={{ url: "/art/awofnbl" }} />);
      expect(wrapper).toMatchSnapshot();
    });

    it ("selects the right item", () => {
        const artWrapper = shallow(<Nav match={{ url: "/art/awofnbl" }} />);
        expect(artWrapper.find("#art.selected")).toHaveLength(1);
        expect(artWrapper.find("#code.selected")).toHaveLength(0);
        expect(artWrapper.find("#about.selected")).toHaveLength(0);

        const codeWrapper = shallow(<Nav match={{ url: "/code/awofnbl" }} />);
        expect(codeWrapper.find("#art.selected")).toHaveLength(0);
        expect(codeWrapper.find("#code.selected")).toHaveLength(1);
        expect(codeWrapper.find("#about.selected")).toHaveLength(0);

        const aboutWrapper = shallow(<Nav match={{ url: "/about/awofnbl" }} />);
        expect(aboutWrapper.find("#art.selected")).toHaveLength(0);
        expect(aboutWrapper.find("#code.selected")).toHaveLength(0);
        expect(aboutWrapper.find("#about.selected")).toHaveLength(1);
    })
})
