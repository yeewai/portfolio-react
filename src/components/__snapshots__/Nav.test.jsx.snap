// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    match={
        Object {
            "url": "/art/awofnbl",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col item selected"
          id="art"
>
          <Link
                    className="nav"
                    replace={false}
                    to="/art"
          >
                    <h2>
                              Art
                    </h2>
          </Link>
</div>,
        <div
          className="col-2 item "
          id="about"
>
          <Link
                    className="nav"
                    replace={false}
                    to="/about"
          >
                    <h2>
                              Yee Wai
                    </h2>
          </Link>
</div>,
        <div
          className="col item "
          id="code"
>
          <Link
                    className="nav"
                    replace={false}
                    to="/code"
          >
                    <h2>
                              Code
                    </h2>
          </Link>
</div>,
      ],
      "className": "mainnav row justify-content-around animated slideInDown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="nav"
            replace={false}
            to="/art"
>
            <h2>
                        Art
            </h2>
</Link>,
          "className": "col item selected",
          "id": "art",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h2>
              Art
</h2>,
            "className": "nav",
            "replace": false,
            "to": "/art",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Art",
            },
            "ref": null,
            "rendered": "Art",
            "type": "h2",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="nav"
            replace={false}
            to="/about"
>
            <h2>
                        Yee Wai
            </h2>
</Link>,
          "className": "col-2 item ",
          "id": "about",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h2>
              Yee Wai
</h2>,
            "className": "nav",
            "replace": false,
            "to": "/about",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yee Wai",
            },
            "ref": null,
            "rendered": "Yee Wai",
            "type": "h2",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="nav"
            replace={false}
            to="/code"
>
            <h2>
                        Code
            </h2>
</Link>,
          "className": "col item ",
          "id": "code",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h2>
              Code
</h2>,
            "className": "nav",
            "replace": false,
            "to": "/code",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Code",
            },
            "ref": null,
            "rendered": "Code",
            "type": "h2",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col item selected"
            id="art"
>
            <Link
                        className="nav"
                        replace={false}
                        to="/art"
            >
                        <h2>
                                    Art
                        </h2>
            </Link>
</div>,
          <div
            className="col-2 item "
            id="about"
>
            <Link
                        className="nav"
                        replace={false}
                        to="/about"
            >
                        <h2>
                                    Yee Wai
                        </h2>
            </Link>
</div>,
          <div
            className="col item "
            id="code"
>
            <Link
                        className="nav"
                        replace={false}
                        to="/code"
            >
                        <h2>
                                    Code
                        </h2>
            </Link>
</div>,
        ],
        "className": "mainnav row justify-content-around animated slideInDown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="nav"
              replace={false}
              to="/art"
>
              <h2>
                            Art
              </h2>
</Link>,
            "className": "col item selected",
            "id": "art",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h2>
                Art
</h2>,
              "className": "nav",
              "replace": false,
              "to": "/art",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Art",
              },
              "ref": null,
              "rendered": "Art",
              "type": "h2",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="nav"
              replace={false}
              to="/about"
>
              <h2>
                            Yee Wai
              </h2>
</Link>,
            "className": "col-2 item ",
            "id": "about",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h2>
                Yee Wai
</h2>,
              "className": "nav",
              "replace": false,
              "to": "/about",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yee Wai",
              },
              "ref": null,
              "rendered": "Yee Wai",
              "type": "h2",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="nav"
              replace={false}
              to="/code"
>
              <h2>
                            Code
              </h2>
</Link>,
            "className": "col item ",
            "id": "code",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h2>
                Code
</h2>,
              "className": "nav",
              "replace": false,
              "to": "/code",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Code",
              },
              "ref": null,
              "rendered": "Code",
              "type": "h2",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
